src/fish_tests.cpp

```diff
@@ -2677,11 +2677,10 @@ static void test_autosuggest_suggest_special() {
         err(L"mkdir failed");  // a path with a double quote
     }
     // This is to ensure tilde expansion is handled. See the `cd ~/test_autosuggest_suggest_specia`
-    // test below. I really dislike this since it mucks with a persons home directory.
-    //
-    // The question is how to modify the test so that tilde expansion can be made hermetic to this
-    // test.
-    if (system("mkdir -p ~/test_autosuggest_suggest_special/")) {
+    // test below.
+    // Fake out the home directory
+    env_set_one(L"HOME", ENV_LOCAL | ENV_EXPORT, L"test/test-home");
```

Comment:
If I understand this, this sets it globally for the remainder of the tests.

I'd either

- reset it

- set it once, at the beginning, for everything

Reply:
OK, I've added a corresponding `env_remove` at the bottom of the function.